compiler/noirc_evaluator/src/brillig/brillig_gen/brillig_block.rs

```diff
@@ -86,8 +86,15 @@ impl<'block, 'global, Registers: RegisterAllocator> BrilligBlock<'block, 'global
         brillig_block.convert_block(dfg);
     }
 
-    pub(crate) fn compile_globals(&mut self, globals: &DataFlowGraph) {
+    pub(crate) fn compile_globals(
+        &mut self,
+        globals: &DataFlowGraph,
+        used_globals: &HashSet<ValueId>,
+    ) {
         for (id, value) in globals.values_iter() {
+            if !used_globals.contains(&id) {
```

Comment:
Seems like we could iterate through used_globals instead which would avoid the need for filtering.

Reply:
That said, this would be a root for indeterminism it seems

Reply:
This comment is mooted if we implement the one below

Reply:
True I could iterate over `used_globals`. I could change it to a `BTreeSet` to avoid indeterminism. Let me know what you prefer.

Reply:
Eh, it's not causing significant slowdown for now. We can clean up later if necessary